File: src/java/ironfist/Lock.java
package ironfist;

/**
 * DOCUMENT ME!
 * 
 * @author pmgremo
 */
public class Lock {
  private Key key;
  private boolean locked;

  /**
   * Creates a new Lock object.
   * 
   * @param key DOCUMENT ME!
   */
  public Lock(Key key) {
    this.key = key;
  }

  /**
   * DOCUMENT ME!
   * 
   * @param key DOCUMENT ME!
   * 
   * @return DOCUMENT ME!
   */
  public boolean unlock(Key key) {
    locked = !this.key.equals(key);

    return !locked;
  }

  /**
   * DOCUMENT ME!
   * 
   * @param key DOCUMENT ME!
   * 
   * @return DOCUMENT ME!
   */
  public boolean lock(Key key) {
    locked = this.key.equals(key);

    return locked;
  }

  /**
   * DOCUMENT ME!
   * 
   * @return DOCUMENT ME!
   */
  public boolean isLocked() {
    return locked;
  }
}

File: src/main/java/ukalus/Lock.kt
package ukalus

/**
 * DOCUMENT ME!

 * @author pmgremo
 */
class Lock
/**
 * Creates a new Lock object.

 * @param key
 * *          DOCUMENT ME!
 */
(private val key: Key) {
    /**
     * DOCUMENT ME!

     * @return DOCUMENT ME!
     */
    var isLocked: Boolean = false
        private set

    /**
     * DOCUMENT ME!

     * @param key
     * *          DOCUMENT ME!
     * *
     * *
     * @return DOCUMENT ME!
     */
    fun unlock(key: Key): Boolean {
        isLocked = this.key != key

        return !isLocked
    }

    /**
     * DOCUMENT ME!

     * @param key
     * *          DOCUMENT ME!
     * *
     * *
     * @return DOCUMENT ME!
     */
    fun lock(key: Key): Boolean {
        isLocked = this.key == key

        return isLocked
    }
}